software/control/core.py

```diff
@@ -860,9 +904,11 @@ def cache_current_position(self):
 
     def move_x(self,delta):
         self.microcontroller.move_x_usteps(int(delta/self.get_mm_per_ustep_X()))
+        self.microcontroller.wait_till_operation_is_completed()
```

Comment:
for now this line may be removed. It it was added, then the self.microcontroller.wait_till_operation_is_completed() needs to be removed after calling `move_x()` in `move_from_click`